urllib3/fields.py

```diff
@@ -130,7 +130,7 @@ def _render_parts(self, header_parts):
             iterable = header_parts.items()
 
         for name, value in iterable:
-            if value:
+            if value or value == '':
```

Comment:
The problem with that logic is that you need to handle *at least* the two empty string types (`b''` and `u''`). More generally, I think the current behaviour (silent non-emission of data for falsy non-strings) is worse than the new failure mode we'd introduce (exceptions thrown for falsy non-strings). We don't accept *non-empty* lists or sets, so I don't see why we should tolerate *empty* lists or sets.